certification/internal/shell/validate_operator_package_name_uniq.go

```diff
@@ -0,0 +1,143 @@
+package shell
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"errors"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+var pyxisBaseUrl = "https://catalog.redhat.com/api/containers/v1/operators/packages"
+
+type packageBody struct {
+	PyxisPackages []PyxisPackages `json:"data"`
+	Page          int             `json:"page"`
+	Page_size     int             `json:"page_size"`
+	Total         int             `json:"total"`
+}
+
+type PyxisPackages struct {
+	Id               string `json:"_id"`
+	Association      string `json:"association"`
+	Create_date      string `json:"creation_date"`
+	Last_update_date string `json:"last_update_date"`
+	Package_name     string `json:"package_name"`
+	Source           string `json:"source"`
+}
+
+type ValidateOperatorPkNameUniqCheck struct {
+}
+
+func (p *ValidateOperatorPkNameUniqCheck) Validate(bundle string) (bool, error) {
+	//log.SetLevel(log.DebugLevel)
+	var bundlePackageName []string
+	var packageBody packageBody
+	var err error
+
+	packageName, err := p.getPackageName(bundle)
+	if err != nil {
+		log.Debug("Error found in getPackaName %s", err)
+		return false, err
+	}
+	//packageName = "argocd-operator"
+	log.Debugf("packagename %s", packageName)
+
+	req, _ := http.NewRequest("GET", pyxisBaseUrl, nil)
+	queryString := req.URL.Query()
+	queryString.Add("filter", fmt.Sprintf("package_name==%s", packageName))
+	req.URL.RawQuery = queryString.Encode()
+	req.Header.Set("X-API-KEY", "RedHatChartVerifier")
+	client := &http.Client{}
+	resp, err := client.Do(req)
```

Comment:
it would be great to merge these lines as much as possible and get rid of the vars that are not reused